### ISSUE TITLE
bug: drawer not not animating in, only out

### ISSUE THREAD

**Issue by Kinbaum** [emilkowalski/vaul#122]
I'm not sure what's going on with this, perhaps you can help me look. I only upgraded from 0.6.5 - 0.6.7 and between these two versions, my drawers are no long sliding up, they just appear. The closing animation works just fine.

I tried replicating this in a sandbox, but I haven't had any luck in doing so. I just lifted the example off your website and ran it in my env and it doesn't animate in anymore.

Any ideas as to where I might be able to look?

https://github.com/emilkowalski/vaul/assets/6590356/cc79a82b-5126-4e8e-9dd9-f5d7fa1262ba

Here is my my two files:

```tsx
// layout.tsx
export default async function RootLayout({ children, params }: {
    children: React.ReactNode;
    params: { locale: Locale };
}) {
    const locale = params.locale;

    return (
        <html
            className={ `motion-safe:!scroll-smooth ${inter.variable}` }
            dir={ locale === 'ar' ? 'rtl' : undefined }
            lang={ locale }
            suppressHydrationWarning>
            <body>
                <main vaul-drawer-wrapper="">{ children }</main>
            </body>
        </html>
    );
}
```

```tsx
// page.tsx
'use client';

import { Drawer } from 'vaul';

export default function HomePage() {
    return (
        <Drawer.Root shouldScaleBackground>
            <Drawer.Trigger asChild>
                <button
                    className="rounded-full bg-white px-4 py-2.5 text-sm font-semibold text-gray-900 shadow-sm ring-1 ring-inset ring-gray-300 hover:bg-gray-50"
                    type="button">
                    Open Drawer
                </button>
            </Drawer.Trigger>
            <Drawer.Portal>
                <Drawer.Overlay className="fixed inset-0 bg-black/40" />
                <Drawer.Content className="fixed inset-x-0 bottom-0 mt-24 flex h-full max-h-[96%] flex-col rounded-t-[10px] bg-gray-100">
                    <div className="flex-1 rounded-t-[10px] bg-white p-4">
                        <div className="mx-auto mb-8 h-1.5 w-12 shrink-0 rounded-full bg-gray-300" />
                        <div className="mx-auto max-w-md">
                            <Drawer.Title className="mb-4 font-medium">Drawer for React.</Drawer.Title>
                            <p className="mb-2 text-gray-600">
                                This component can be used as a Dialog replacement on mobile and tablet devices.
                            </p>
                            <p className="mb-2 text-gray-600">
                                It comes unstyled, has gesture-driven animations, and is made by{ ' ' }
                                <a
                                    className="underline"
                                    href="https://emilkowal.ski/"
                                    target="_blank">
                                    Emil Kowalski
                                </a>
                                .
                            </p>
                            <p className="mb-8 text-gray-600">
                                It uses{ ' ' }
                                <a
                                    className="underline"
                                    href="https://www.radix-ui.com/docs/primitives/components/dialog"
                                    target="_blank">
                                    Radix&apos;s Dialog primitive
                                </a>{ ' ' }
                                under the hood and is inspired by{ ' ' }
                                <a
                                    className="underline"
                                    href="https://twitter.com/devongovett/status/1674470185783402496"
                                    target="_blank">
                                    this tweet.
                                </a>
                            </p>
                        </div>
                    </div>
                    <div className="mt-auto border-t border-gray-200 bg-gray-100 p-4">
                        <div className="mx-auto flex max-w-md justify-end gap-6">
                            <a
                                className="gap-0.25 flex items-center text-xs text-gray-600"
                                href="https://github.com/emilkowalski/vaul"
                                target="_blank">
                                GitHub
                                <svg
                                    aria-hidden="true"
                                    className="ml-1 h-3 w-3"
                                    fill="none"
                                    height="16"
                                    stroke="currentColor"
                                    strokeLinecap="round"
                                    strokeLinejoin="round"
                                    strokeWidth="2"
                                    viewBox="0 0 24 24"
                                    width="16">
                                    <path d="M18 13v6a2 2 0 01-2 2H5a2 2 0 01-2-2V8a2 2 0 012-2h6" />
                                    <path d="M15 3h6v6" />
                                    <path d="M10 14L21 3" />
                                </svg>
                            </a>
                            <a
                                className="gap-0.25 flex items-center text-xs text-gray-600"
                                href="https://twitter.com/emilkowalski_"
                                target="_blank">
                                Twitter
                                <svg
                                    aria-hidden="true"
                                    className="ml-1 h-3 w-3"
                                    fill="none"
                                    height="16"
                                    stroke="currentColor"
                                    strokeLinecap="round"
                                    strokeLinejoin="round"
                                    strokeWidth="2"
                                    viewBox="0 0 24 24"
                                    width="16">
                                    <path d="M18 13v6a2 2 0 01-2 2H5a2 2 0 01-2-2V8a2 2 0 012-2h6" />
                                    <path d="M15 3h6v6" />
                                    <path d="M10 14L21 3" />
                                </svg>
                            </a>
                        </div>
                    </div>
                </Drawer.Content>
            </Drawer.Portal>
        </Drawer.Root>
    );
}
```

**Comment by emilkowalski** [comment]
Looks pretty normal, maybe try removing all the attributes on the `html` tag and see if it does anything? If that won't help then I'd need a reproduction on codesandbox unfortunately.

**Comment by Kinbaum** [comment]
So I cloned the repo locally and moved the src files into my project. Still encountered the same issue as downloading from npm. I was able to tinker with it enough to figure out what would allow it to work again. Let me know your thoughts.

In the `Content` I ended up having to restore the following effect:

```tsx
React.useEffect(() => {
    // Trigger enter animation without using CSS animation
    setVisible(true);
}, []);
```

Then, in the `onRelease` function, I had to modify the effect to the following:

```tsx
React.useEffect(() => {
    // Trigger enter animation without using CSS animation
    if (isOpen) {
        openTime.current = new Date();
        scaleBackground(true);
    }
}, [ isOpen ]);
```

These two changes restored the animation on mount. I have no idea as to why this started happening all of a sudden and I can't replicate it using a vanilla next app. Wonder if it has something to do with app complexity.
